internal/literals/asthelpers.go

```diff
@@ -0,0 +1,77 @@
+package literals
+
+import (
+	"go/ast"
+	"go/token"
+)
+
+func ident(name string) *ast.Ident {
+	return &ast.Ident{Name: name}
+}
+func intLiteral(value string) *ast.BasicLit {
+	return &ast.BasicLit{
+		Kind:  token.INT,
+		Value: value,
+	}
+}
+
+// name[index]
+func indexExpr(name string, index ast.Expr) *ast.IndexExpr {
+	return &ast.IndexExpr{
+		X:     ident(name),
+		Index: index,
+	}
+}
+
+// convertType(arg)
+func convertExpr(convertType string, arg ast.Expr) *ast.CallExpr {
+	return &ast.CallExpr{
+		Fun:  ident(convertType),
+		Args: []ast.Expr{arg},
+	}
+}
+
+// fun(arg)
+func callExpr(fun ast.Expr, arg ast.Expr) *ast.CallExpr {
+	var args []ast.Expr
+	if arg != nil {
+		args = []ast.Expr{arg}
+	}
+
+	return &ast.CallExpr{
+		Fun:  fun,
+		Args: args,
+	}
+}
+
+// func() resultType {block}()
+func lambdaCall(resultType ast.Expr, block *ast.BlockStmt) *ast.CallExpr {
+	funcLit := &ast.FuncLit{
+		Type: &ast.FuncType{
+			Params: &ast.FieldList{},
+			Results: &ast.FieldList{
+				List: []*ast.Field{
+					{Type: resultType},
+				},
+			},
+		},
+		Body: block,
+	}
+	return callExpr(funcLit, nil)
+}
+
+// return result
+func returnStmt(result ast.Expr) *ast.ReturnStmt {
```

Comment:
easier to make this `results ...ast.Expr`

Reply:
Right